test/gpdb/scripts/test_pitr_gpdb6.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+set -exo pipefail
+
+PGBACKREST_TEST_DIR=/home/gpadmin/test_pgbackrest
+PGBACKREST_BIN=/usr/local/bin
+GPHOME=/usr/local/greenplum-db-devel
+
+# Starting up demo cluster
+source "$GPHOME/greenplum_path.sh"
+pushd gpdb_src/gpAux/gpdemo
+make create-demo-cluster WITH_MIRRORS=true
+source gpdemo-env.sh
+popd
+
+# Creating backup and log directories for pgbackrest
+TEST_NAME=$(basename "${0%.sh}")
+mkdir -p "$PGBACKREST_TEST_DIR/logs/$TEST_NAME"
+mkdir -p "$PGBACKREST_TEST_DIR/$TEST_NAME"
+DATADIR="${MASTER_DATA_DIRECTORY%*/*/*}"
+MASTER=${DATADIR}/qddir/demoDataDir-1
+PRIMARY1=${DATADIR}/dbfast1/demoDataDir0
+PRIMARY2=${DATADIR}/dbfast2/demoDataDir1
+PRIMARY3=${DATADIR}/dbfast3/demoDataDir2
+MIRROR1=${DATADIR}/dbfast_mirror1/demoDataDir0
+MIRROR2=${DATADIR}/dbfast_mirror2/demoDataDir1
+MIRROR3=${DATADIR}/dbfast_mirror3/demoDataDir2
+
+# Filling the pgbackrest.conf configuration file
+cat <<EOF > $PGBACKREST_TEST_DIR/pgbackrest.conf
+[seg-1]
+pg1-path=$MASTER
+pg1-port=6000
+
+[seg0]
+pg1-path=$PRIMARY1
+pg1-port=6002
+
+[seg1]
+pg1-path=$PRIMARY2
+pg1-port=6003
+
+[seg2]
+pg1-path=$PRIMARY3
+pg1-port=6004
+
+[global]
+repo1-path=$PGBACKREST_TEST_DIR/$TEST_NAME
+log-path=$PGBACKREST_TEST_DIR/logs/$TEST_NAME
+start-fast=y
+fork=GPDB
+EOF
+
+# Initializing pgbackrest for GPDB
+for i in -1 0 1 2
+do 
+    PGOPTIONS="-c gp_session_role=utility" $PGBACKREST_BIN/pgbackrest \
+    --config $PGBACKREST_TEST_DIR/pgbackrest.conf --stanza=seg$i stanza-create
+done
+
+# Configuring WAL archiving command
+gpconfig -c archive_mode -v on
+
+gpconfig -c archive_command -v "'PGOPTIONS=\"-c gp_session_role=utility\" \
+$PGBACKREST_BIN/pgbackrest --config $PGBACKREST_TEST_DIR/pgbackrest.conf \
+--stanza=seg%c archive-push %p'" --skipvalidation
+
+gpstop -ar
+
+# pgbackrest health check
+for i in -1 0 1 2
+do 
+    PGOPTIONS="-c gp_session_role=utility" $PGBACKREST_BIN/pgbackrest \
+    --config $PGBACKREST_TEST_DIR/pgbackrest.conf --stanza=seg$i check
+done
+
+# The test scenario starts here
+# Creating initial dataset
+export PGDATABASE=gpdb_pitr_database
+createdb
+psql -c "CREATE TABLE t1 (id int, text varchar(255)) DISTRIBUTED BY (id);"
```

Comment:
What about one CTAS query instead of CREATE TABLE and INSERT?
```sql
create table t1 as select id, 'text'||id text from generate_series(1,30) id  distributed by (id);
```